src/main/java/org/iotdata/constants/dml/TagsQueries.java

```diff
@@ -44,27 +86,43 @@ public class TagsQueries {
 			?resultNode a msr:Measure ;
 				msr:Unit aiotp2:bpm ;
 				msr:hasNumericalValue ?heartRate .
+			
+			FILTER (?heartRate != 0)
 			}
+			GROUP BY ?identifier
 			""";
 
-	public static final String SELECT_DAYS_WITH_WATCH_BATTERY_LEVELS = """
-			SELECT ?timeStamp ?batteryLevel ?identifier
+	public static final String SELECT_ALARM_EVENTS  = """
+			SELECT ?identifier
+					?startTime
+					(?timeStamp AS ?finishTime)
+					?duration
 			WHERE {
-			?subject schema:identifier ?identifier ;
-				sosa:hosts ?sensor .
-				
-			?sensor sosa:madeObservation ?observation .
-			
-			?observation a sosa:Observation ;
-				sosa:resultTime ?timeStamp ;
-				sosa:hasResult ?resultNode .
-				
-			?resultNode a aiotp2:TagMetadataResult ;
-				aiotp2:hasWatchBatteryLevel ?batteryLevel .
+				{
+					SELECT ?timeStamp
+					   	   ?alarmStatus
+					       ?identifier
+					WHERE {
+					?subject schema:identifier ?identifier ;
+						sosa:hosts ?sensor .
+
+					?sensor sosa:madeObservation ?observation .
+
+					?observation a sosa:Observation ;
+						sosa:resultTime ?timeStamp ;
+						sosa:hasResult ?resultNode .
+						
+					?resultNode a aiotp2:TagMetadataResult ;
+						aiotp2:hasAlarm ?alarmStatus .
+					}
+				}
+				BIND(func:mapAlarmEvent(?identifier, ?timeStamp, ?alarmStatus) as ?startTime)
+				BIND((?timeStamp - ?startTime) as ?duration)
+				FILTER(?startTime != ?timeStamp && day(?duration) < 1 && SUBSTR(STR(?duration), 1, 3) != "-PT")
 			}
 			""";
 
-	public static final String SELECT_DAYS_WITH_ALARM = """
+	public static final String SELECT_COUNT_ALARM_TRIGGERS = """
```

Comment:
I wanted to return all timestamps in which the alarm was triggered and the associated identifier, was also thinking about counting but wasn't sure if it will lead to the good result, as alarm triggers at different timestamps can refer to the same alarm event, to count the I was thinking to use SELECT_ALARM_EVENTS query.

Reply:
Yeah but I was thinking more about what this query can give us :) I'll still have to do some operations on that on the PowerBi, so if there is a possibility to simplify it here, then it would be much more efficient.

Reply:
I mean, could you please give me an example on what information I can draw from the results of this query?